File: Python/FilmBioskop.py
import sys, os
sys.path.insert(0, os.path.abspath(os.path.dirname(__file__)))
from Film import Film


class FilmBioskop(Film):
    def __init__(self, id=0, judul="", genre="", durasi=0, studio="", klasifikasiUsia=""):
        super().__init__(id, judul, genre)
        self.__durasi = durasi
        self.__studio = studio
        self.__klasifikasiUsia = klasifikasiUsia

    # Getter
    def getDurasi(self):
        return self.__durasi

    def getStudio(self):
        return self.__studio

    def getKlasifikasiUsia(self):
        return self.__klasifikasiUsia

    # Setter
    def setDurasi(self, durasi):
        self.__durasi = durasi

    def setStudio(self, studio):
        self.__studio = studio

    def setKlasifikasiUsia(self, klasifikasiUsia):
        self.__klasifikasiUsia = klasifikasiUsia

File: CPP/FilmBioskop.cpp
#include <string>

#include "Film.cpp"

using namespace std;

class FilmBioskop : public Film {
private:
    int durasi;
    string studio;
    string klasifikasiUsia;

public:
    // Constructor default
    FilmBioskop() : Film() {
        durasi = 0;
        studio = "";
        klasifikasiUsia = "";
    }

    // Constructor berparameter
    FilmBioskop(
        int id,
        string judul,
        string genre,
        int durasi,
        string studio,
        string klasifikasiUsia
    ) : Film(id, judul, genre) {

        this->durasi = durasi;
        this->studio = studio;
        this->klasifikasiUsia = klasifikasiUsia;
    }

    // Getter
    int getDurasi() const {
        return durasi;
    }

    string getStudio() const {
        return studio;
    }

    string getKlasifikasiUsia() const {
        return klasifikasiUsia;
    }

    // Setter
    void setDurasi(int durasi) {
        this->durasi = durasi;
    }

    void setStudio(string studio) {
        this->studio = studio;
    }

    void setKlasifikasiUsia(string klasifikasiUsia) {
        this->klasifikasiUsia = klasifikasiUsia;
    }

    // Destructor
    ~FilmBioskop() override {
    }
};